testing/middleware/builder/builder.go

```diff
@@ -576,12 +595,148 @@ func (p *Builder) handleHeaderRequestDeneb(w http.ResponseWriter) {
+	if err != nil {
+		p.cfg.logger.WithError(err).Error("Could not get decoded execution requests")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	rv1 := &builderAPI.ExecutionRequestsV1{
+		Deposits:       make([]*builderAPI.DepositRequestV1, len(requests.Deposits)),
+		Withdrawals:    make([]*builderAPI.WithdrawalRequestV1, len(requests.Withdrawals)),
+		Consolidations: make([]*builderAPI.ConsolidationRequestV1, len(requests.Consolidations)),
+	}
+
+	for i, d := range requests.Deposits {
+		amount := new(big.Int).SetUint64(d.Amount)
+		index := new(big.Int).SetUint64(d.Index)
+		dr := &builderAPI.DepositRequestV1{
+			PubKey:                d.Pubkey,
+			WithdrawalCredentials: d.WithdrawalCredentials,
+			Amount:                builderAPI.Uint256{Int: amount},
+			Signature:             d.Signature,
+			Index:                 builderAPI.Uint256{Int: index},
+		}
+		rv1.Deposits[i] = dr
+	}
+
+	for i, w := range requests.Withdrawals {
+		bi := new(big.Int).SetUint64(w.Amount)
+		wr := &builderAPI.WithdrawalRequestV1{
+			SourceAddress:   w.SourceAddress,
+			ValidatorPubkey: w.ValidatorPubkey,
+			Amount:          builderAPI.Uint256{Int: bi},
+		}
+		rv1.Withdrawals[i] = wr
+	}
+
+	for i, c := range requests.Consolidations {
+		cr := &builderAPI.ConsolidationRequestV1{
+			SourceAddress: c.SourceAddress,
+			SourcePubkey:  c.SourcePubkey,
+			TargetPubkey:  c.TargetPubkey,
+		}
+		rv1.Consolidations[i] = cr
+	}
+
+	bid := &builderAPI.BuilderBidElectra{
+		Header:             wrappedHdr,
+		BlobKzgCommitments: commitments,
+		Value:              val,
+		Pubkey:             secKey.PublicKey().Marshal(),
+		ExecutionRequests:  rv1,
+	}
+
+	sszBid := &eth.BuilderBidElectra{
+		Header:             hdr,
+		BlobKzgCommitments: b.BlobsBundle.KzgCommitments,
+		Value:              val.SSZBytes(),
+		Pubkey:             secKey.PublicKey().Marshal(),
+		ExecutionRequests:  requests,
+	}
+	d, err := signing.ComputeDomain(params.BeaconConfig().DomainApplicationBuilder,
+		nil, /* fork version */
+		nil /* genesis val root */)
+	if err != nil {
+		p.cfg.logger.WithError(err).Error("Could not compute the domain")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	rt, err := signing.ComputeSigningRoot(sszBid, d)
+	if err != nil {
+		p.cfg.logger.WithError(err).Error("Could not compute the signing root")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	sig := secKey.Sign(rt[:])
+	hdrResp := &ExecHeaderResponseElectra{
+		Version: "electra",
+		Data: struct {
+			Signature hexutil.Bytes                 `json:"signature"`
+			Message   *builderAPI.BuilderBidElectra `json:"message"`
+		}{
+			Signature: sig.Marshal(),
+			Message:   bid,
+		},
+	}
+
+	err = json.NewEncoder(w).Encode(hdrResp)
+	if err != nil {
+		p.cfg.logger.WithError(err).Error("Could not encode response")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	p.currVersion = version.Electra
```

Comment:
i'm not sure if this is the best way to do it, it's because the payload is still deneb so I can't do it by payload version